### ISSUE TITLE
Dont work after 100+ process?

### ISSUE THREAD

**Issue by shirshak55** [laravel-zero/laravel-zero#206]
I just made a simple bot it seems i cannot go beyond 100+ process on windows cli? the process created after 100+ is slow. Do you have any idea where this magic number 100 is coming thanks.

Using latest version of laravel zero. I wonder if its php cli on windows issue :(

**Comment by juniorb2ss** [comment]
@shirshak55

You tried use docker (with UNIX distribution) to see if has difference? How much you have memory, and, how much you specific for each worker?

**Comment by shirshak55** [comment]
@juniorb2ss 64gb ram so no problem in ram. And i have set to 20gb in php.ini files.   Hmm no i designed it for windows so i cannot install dockers etc on others computer.

**Comment by juniorb2ss** [comment]
@shirshak55 You using queue or calling command directly? You use any external service? e.g: redis, mysql, memcached and etc?

**Comment by shirshak55** [comment]
just use like 

open 100 command prompt and run the app. 

1 start the app
2 start the app
3 start the app
4 start the app
5 start the app
...
...
100 start the app
101 start the app
102 start the app

101  and 102 don't work or works too slow . And if i lets say close 2. 101 suddenly starts to work properly.


The only thing is do is use guzzle for api request nothing more than that.

**Comment by juniorb2ss** [comment]
I do not see any limitation on the FW. 

You said that you consume an API, have you checked if the API has limits for simultaneous requests? Did you try to debug somehow to see where the slow is? I recommend using blackfire for example, enabling extension when you run workers 100+.

**Comment by shirshak55** [comment]
no api has no limit because i use proxy. When i open same api from browser it shows.
100 is definitely coming from somewhere because this problem starts at 100 only.

hmm never heard of backfire.

is there any limitation on os how much process php cli can spawn?

**Comment by juniorb2ss** [comment]
@shirshak55 

No. PHP Cli don't have limitation about instances. Only limitations is your process number (cores) and memory limit.

You said API dont't has limitation, ok... but your proxy maybe have? I recommend to check everything, because php or fw no has limitation.

Unfortunately I do not have windows, otherwise I would create a basic job and try to simulate the amount of workers.

If you can not find the reason, I recommend creating a basic job, such as a hello-world, and this job, you can scale it to 100+ to see if the same problem occurs.

If it does not, it may be a sign that the problem is elsewhere than php/fw

**Comment by shirshak55** [comment]
can u say me related to process number? and how do i know if cores are limiting it?

**Comment by juniorb2ss** [comment]
I recommend to read: https://haydenjames.io/php-performance-additional-cpu-cores-vs-faster-cpu-cores/

https://medium.com/@rossbulat/true-php7-multi-threading-how-to-rebuild-php-and-use-pthreads-bed4243c0561

But I think the problem is not in memory/processor. Because the number of limited workers is always the same... 100+.

That is, the problem is another limitation in the proxy, API, IO, MySQL etc.
